### PR TITLE
Implement `summarize_story` for web

### LINKED ISSUE #28: Implement `summarize_story` for web
The `summarize_story` task type needs to display these fields:
*  `story`

It takes the following interaction type:
*  `type`: `text_reply_to_post`
*  `text`: The user's written summary

### PR DESCRIPTION
**Pull request by AbdBarho**
Partially closes #28

There is no corresponding endpoint for submission the backend, so the submission is left unimplemented.

Merry Christmas 🎄

### CODE REVIEW

File: website/src/pages/summarize/story.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      console.log(data);
+      setTasks([data]);
+    },
+  });
+
+  // Trigger a mutation that updates the current task.  We should probably
+  // signal somewhere that this interaction is being processed.
+  const submitResponse = (task: { id: string }) => {
+    const text = inputRef.current.value.trim();
+    // TODO: implement after we have the backend endpoint
```
Comment (fozziethebeat) [inline website/src/pages/summarize/story.tsx:27]: can you uncomment and implement the mutation code here (mostly copied from `website/src/pages/grading/grade-output.tsx`) with a new field in the trigger body (that i'll make work in a following PR): `update_type: "text_reply_to_post"`

I can make this work pretty quick.

File: website/src/pages/summarize/story.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
+import { Textarea } from "@chakra-ui/react";
```
Comment (fozziethebeat) [inline website/src/pages/summarize/story.tsx:1]: Can you add a comment at the top here:

`// TODO(#65): Unify and simplify the task paths.`

**Review by fozziethebeat — APPROVED** [review]
If you can, please make the two small requested changes, otherwise I'll make them in a follow up PR.

File: website/src/pages/summarize/story.tsx (hunk 1 above)
Comment (AbdBarho) [reply website/src/pages/summarize/story.tsx:27]: @fozziethebeat gladly, quick question: `post` in the sense of HTTP POST, or post as in reddit post?

**Comment by andreaskoepf** [comment]
> There is no corresponding endpoint for submission the backend, so the submission is left unimplemented.

Merry X-Mas! :-) The endpoint to submit user generated summaries is the POST `/api/v1/tasks/interaction` endpoint, please see the text-frontend [here](https://github.com/LAION-AI/Open-Assistant/blob/f8c300800326f27180197ce5a7270c7ef3c9b9b2/text-frontend/__main__.py#L54-L63) and the discord-bot [ApiClient](https://github.com/LAION-AI/Open-Assistant/blob/f8c300800326f27180197ce5a7270c7ef3c9b9b2/discord-bot/api_client.py#L72-L74) and [SummarizeStoryHandler](https://github.com/LAION-AI/Open-Assistant/blob/f8c300800326f27180197ce5a7270c7ef3c9b9b2/discord-bot/task_handlers.py#L117-L130).

File: website/src/pages/summarize/story.tsx (hunk 1 above)
Comment (andreaskoepf) [reply website/src/pages/summarize/story.tsx]: post in `text_reply_to_post` refers to a row in the database in the post table. The post table allows to store conversation trees (threads or completions of different users or model sampling runs). The interactions like ratings and rankings are store in a separate table called post_reaction (currently users can only submit a single reaction to a given post).

**Comment by AbdBarho** [comment]
@andreaskoepf thank you, I think we already have a backend that calls the right api:
https://github.com/LAION-AI/Open-Assistant/blob/54c7a942acdc2a438c73f6b1c1eec2555832c8f6/website/src/pages/api/update_task.ts#L42-L59

This was a bit unclear for me at the beginning with 2 backends, it is still unclear to me why it has to be like this, but for an MVP its ok.

**Review by yk — APPROVED** [review]
thank you very much :)

just briefly regarding the "two backends". The real backend is the fastapi backend. The nextjs project is technically considered a "frontend" (even though it consists of its own back- and frontend). Other frontends exist, such as the discord bot. Frontends are responsible for making the user interaction happen and then sending all information to the backend.

The architecture is like this so we don't have to re-implement a lot of logic in all the place, like the bot, the website, etc.

**Comment by fozziethebeat** [comment]
There's a small change in the trigger body that I'll make in a follow up PR.

**Pull request merged**